src/java_binding/src/iterator.rs

```diff
@@ -159,26 +165,40 @@ impl Iterator {
 
         let stream = select_all_vnode_stream(streams);
 
-        Ok(Self {
-            row_serializer: RowDeserializer::new(
-                read_plan
-                    .table_catalog
-                    .unwrap()
-                    .columns
-                    .into_iter()
-                    .map(|c| (&c.column_desc.unwrap().column_type.unwrap()).into())
-                    .collect(),
-            ),
-            stream,
-        })
+        let table = read_plan.table_catalog.unwrap();
+        let versioned = table.version.is_some();
+        let (column_ids, schema): (Vec<_>, Vec<_>) = table
+            .columns
+            .into_iter()
+            .map(|c| c.column_desc.unwrap())
+            .map(|c| {
+                (
+                    ColumnId::new(c.column_id),
+                    DataType::from(&c.column_type.unwrap()),
+                )
+            })
+            .unzip();
+
+        // Decide which serializer to use based on whether the table is versioned or not.
+        let row_serde = if versioned {
+            ColumnAwareSerde::new(&column_ids, schema.into()).into()
+        } else {
+            BasicSerde::new(&column_ids, schema.into()).into()
+        };
```

Comment:
I just found that there's something else we need to update in the storage java binding. cc @wenym1 Is it possible to reuse some code of `StorageTable` here?